user_connections.go

```diff
@@ -0,0 +1,161 @@
+	getActiveConnectionsCount() int
+	getNick() string
+}
+
+// ConnectedUsersPool is responsible for tracking all user handlers
+type ConnectedUsersPool struct {
+	handlers map[string]SockchatUserHandler
+	lock     sync.RWMutex
+}
+
+func NewUserManager() *ConnectedUsersPool {
+	manager := &ConnectedUsersPool{
+		handlers: make(map[string]SockchatUserHandler),
+	}
+	return manager
+}
+
+func (m *ConnectedUsersPool) GetHandler(nick string) (SockchatUserHandler, bool) {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	handler, ok := m.handlers[nick]
+	return handler, ok
+}
+
+func (m *ConnectedUsersPool) AddConnection(conn *SockChatWS, store SockchatChannelStore) SockchatUserHandler {
+	handler, ok := m.GetHandler(conn.Nick)
+	if !ok {
+		handler = NewUserHandler(conn, store)
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		m.handlers[conn.Nick] = handler
+	}
+	handler.AddConnection(conn)
+	return handler
+}
+
+func (m *ConnectedUsersPool) RemoveConnection(conn *SockChatWS) {
+	handler, ok := m.GetHandler(conn.Nick)
+	if !ok {
+		log.Printf("could not drop connection - no handler found for %s", conn.Nick)
+		return
+	}
+
+	if handler.getActiveConnectionsCount() > 1 {
+		handler.RemoveConnection(conn)
+	} else {
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		delete(m.handlers, conn.Nick)
+	}
+}
+
+// UserHandler manages connections of a single connected user
+type UserHandler struct {
+	Nick         string
+	connections  map[*SockChatWS]bool
+	requests     chan *UserHandlerRequest
+	lock         sync.RWMutex
+	channelStore SockchatChannelStore
+}
+
+type UserHandlerRequest struct {
+	action      string
+	payload     any
+	errCallback chan error
+}
+
+func NewUserHandler(conn *SockChatWS, store SockchatChannelStore) *UserHandler {
+	handler := UserHandler{
+		Nick:         conn.Nick,
+		connections:  make(map[*SockChatWS]bool),
+		requests:     make(chan *UserHandlerRequest),
+		channelStore: store,
+	}
+	go handler.HandleRequests()
+	return &handler
+}
+
+func (u *UserHandler) HandleRequests() {
+	for {
+		req := <-u.requests
+		switch req.action {
+		case CreateAction:
+			err := u.channelStore.CreateChannel(req.payload.(*ChannelRequest).Name)
+			if err != nil {
+				req.errCallback <- err
+				continue
+			}
+			req.errCallback <- u.channelStore.AddUserToChannel(req.payload.(*ChannelRequest).Name, u)
+		case JoinAction:
+			err := u.channelStore.AddUserToChannel(req.payload.(*ChannelRequest).Name, u)
+			req.errCallback <- err
+		case LeaveAction:
+			err := u.channelStore.RemoveUserFromChannel(req.payload.(*ChannelRequest).Name, u)
+			req.errCallback <- err
+		case SendMessageAction:
+			channel, err := u.channelStore.GetChannel(req.payload.(*SendMessageRequest).Channel)
+			if !channel.GetMembers()[u] {
+				req.errCallback <- ErrUserNotInChannel
+				continue
+			}
+			req.errCallback <- err
+			channel.MessageMembers(NewSocketMessage("new_message", MessageEvent{req.payload.(*SendMessageRequest).Text, req.payload.(*SendMessageRequest).Channel, u.Nick}))
```

Comment:
```suggestion
			channel.MessageMembers(NewSocketMessage("new_message", MessageEvent{req.payload.(*SendMessageRequest).Text, req.payload.(*SendMessageRequest).Channel, u.getNick()}))
```
Nick is fetched using the `getNick` method except in here

Reply:
changed, but  does it matter here? It's method of UserHandler anyway, I guess it's just for consistency?

Reply:
Just for consistency and code flexibility - if you happen to change `getNick` to `getUserConnectionInfo` it won't be reflected everywhere